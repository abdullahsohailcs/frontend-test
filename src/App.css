.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

input:focus-visible {
  border: 3px solid #6833FF;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* src/styles/responsive.css */
.responsive-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
}

.timer-container,
.dc-container {
  width: auto;  /* Keep them auto for default layout */
}

/* For smaller screens, stack the elements */
@media (max-width: 350px) {
  .responsive-container {
    flex-direction: column;
  }

  .timer-container,
  .dc-container {
    margin-bottom: 10px;  /* Space between stacked items */
  }

  .dc-container {
    text-align: center; /* Center the 'DC' text when stacked */
  }
}
