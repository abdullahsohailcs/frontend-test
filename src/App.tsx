import React from 'react';
import SearchComponent from './components/SearchComponent';
import './App.css';
import Timer from './components/timer';
import Logo from './components/svg/logo';

const App: React.FC = () => {
  return (
    <div className="min-h-screen flex flex-col items-center justify-center"
      style={{ background: '#EDF2F7', paddingTop: '142px' }}>

      <SearchComponent />

      {/* DC and Timer positioned at the extreme ends of the screen */}
      <div
        className="flex items-center justify-between w-full responsive-container"
        style={{
          maxWidth: '100%',
          padding: '48px 50px 49.72px 50px',
        }}
      >
        {/* Timer on the left */}
        <div className="timer-container">
          <Timer />
        </div>

        {/* DC on the right */}
        <div
          style={{
            width: '82px',
            height: '44.28px',
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
          }}
          className="dc-container"
        >
          <Logo></Logo>
        </div>
      </div>
    </div>
  );
};

export default App;