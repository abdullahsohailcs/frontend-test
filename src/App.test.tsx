import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

// Mocking child components for simplicity in App testing
jest.mock('./components/SearchComponent', () => () => <div>Search Component</div>);
jest.mock('./components/timer', () => () => <div>Timer Component</div>);

describe('App Component', () => {
  it('renders correctly and matches snapshot', () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });
});
