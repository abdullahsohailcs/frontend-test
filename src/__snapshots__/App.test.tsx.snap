// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen flex flex-col items-center justify-center"
    style="background: rgb(237, 242, 247); padding-top: 140px;"
  >
    <div>
      Search Component
    </div>
    <div
      class="flex items-center justify-between w-full mt-4"
      style="max-width: 100%; padding: 48px 49.72px 49.72px 49.72px;"
    >
      <div>
        Timer Component
      </div>
      <div
        style="width: 82px; height: 44.28px; display: flex; align-items: center; justify-content: center; color: rgb(213, 217, 222); font-size: 18px; font-weight: bold;"
      >
        DC
      </div>
    </div>
  </div>
</DocumentFragment>
`;
